teachopencadd/talktorials/T018_automated_cadd_pipeline/utils/helpers/dogsitescorer.py

```diff
@@ -0,0 +1,357 @@
+    return
+
+
+def select_best_pocket(binding_site_df, selection_method, selection_criteria, ascending=False):
+    """
+    Select the best binding site from the table of all detected binding sites,
+    either by sorting the binding sites based on a set of properties in the table,
+    or by applying a function on the property values.
+
+    Parameters
+    ----------
+    binding_site_df : pandas.DataFrame
+        Binding site data retrieved from the DoGSiteScorer webserver.
+    selection_method : str
+        Selection method for selecting the best binding site.
+        Either 'sorting' or 'function'.
+    selection_criteria : str or list
+        If 'selection_method' is 'sorting':
+            List of one or more property names.
+        If 'selection_method' is 'function':
+            Any valid python syntax that generates a list-like object
+            with the same length as the number of detected binding sites.
+    ascending : bool
+        Optional; default: False.
+        If set to True, the binding site with the lowest value will be selected,
+        otherwise, the binding site with the highest value is selected.
+
+    Returns
+    -------
+    str
+        Name of the selected binding site.
+    """
+    df = binding_site_df
+
+    if selection_method == "sorting":
+        sorted_df = df.sort_values(by=selection_criteria, ascending=ascending)
+    elif selection_method == "function":
+        df["function_score"] = eval(selection_criteria)
+        sorted_df = df.sort_values(by="function_score", ascending=ascending)
+    else:
+        raise ValueError(f"Binding site selection method unknown: {selection_method}")
+
+    selected_pocket_name = sorted_df.iloc[0].name
+    return selected_pocket_name
+
+
+def calculate_pocket_coordinates_from_pocket_pdb_file(filepath):
+    """
+    Calculate the coordinates of a binding site using the binding site's PDB file
+    downloaded from DoGSiteScorer.
+
+    Parameters
+    ----------
+    filepath : str or pathlib.Path
+        Local filepath (including filename, without extension) of the binding site's PDB file.
+
+    Returns
+    -------
+    dict of list of int
+        Binding site coordinates in format:
+        `{'center': [x, y, z], 'size': [x, y, z]}`
+    """
+    with open(f"{filepath}.pdb") as f:
+        pdb_file_text_content = f.read()
+    pdb_file_df = pdb.load_pdb_file_as_dataframe(pdb_file_text_content)
+    pocket_coordinates_data = pdb_file_df["OTHERS"].loc[5, "entry"]
+    coordinates_data_as_list = pocket_coordinates_data.split(" ")
+    # `coordinates_data_as_list` looks like this:
+    # ['\tGeometric', 'pocket', 'center', 'at', '', '15.91', '', '32.33', '', '11.03', 'with', 'max', 'radius', '12.42']
+
+    # Let's keep only coordinates
+    coordinates = []
+    for elem in coordinates_data_as_list:
+        try:
+            # If element can be cast to float (i.e. is a coordinate)
+            coordinates.append(float(elem))
+        except:
```

Comment:
Catch TypeError and ValueError here.

Reply:
That's not what we want to do here (see my comment above: https://github.com/volkamerlab/teachopencadd/pull/179#discussion_r749583197). Basically we have a list of strings, some of which represent floats, and we want to extract those floats from the list only. So we iterate over the list and `try` to cast each element into float; if it's not possible, that means we don't want that element and thus we `pass` and go to the next element.

StackOverflow suggests that's the most efficient and pythonic way to do it (if we don't want to import `re` just to do this). However, if you don't like the `try, except: pass` solution, I could also do something like this:

```python
for elem in coordinates_data_as_list:
        # remove only one '.' and check if the result is an integer
        if elem.replace('.','',1).isdigit():
            coordinates.append(float(elem))
```
Or do you have any better solutions @jaimergp?

Reply:
went with the regex solution (see 12f9db2):
```python
coordinates = [float(element) for element in coordinates_data_as_list if re.compile(r'\d+(?:\.\d*)').match(element)]
```